### PR TITLE
[NPU] bugs fix for Deepseek models 

### PR DESCRIPTION
**Pull request by MichelleWu351**
## Motivation

When we run DeepSeek V3.1 and DeepSeek V3.2 models on Ascend NPU devices, we encounter three issues. 
1.  when we export SGLANG_NPU_USE_MLAPO=1, this feature does not work.
2. The error message is `Tensor self not implemented for DT_UINT64`.  It is about the data type of `bootstrap_room` in utils.py. 
image

## Modifications

To fix three bugs: 
1. We corrected the  config name from `quant_config` to` quantization_config` in mla_preprocess.py, so when we export SGLANG_NPU_USE_MLAPO=1, this feature can work.
2. The bug about data type of `bootstrap_room` in utils.py is fixed by `https://github.com/sgl-project/sglang/pull/19423`.
In addition，we reordered a `prefill_delayer_single_pass` check in the scheduling policy to enable an earlier exit and improve performance.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).2

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
4. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
5. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py [hunk 1]
```diff
@@ -459,8 +459,9 @@ def forward(self, positions, hidden_states, forward_batch, zero_allocator):
         # assert self.quant_config and self.quant_config.get_name() == "modelslim"
         # route by `qkv_a_proj` quant type as MTP layers can be unquantized
         _is_w8a8 = (
-            hasattr(self.qkv_a_proj.quant_method, "quant_config")
-            and self.qkv_a_proj.quant_method.quant_config.get_name() == "modelslim"
+            hasattr(self.qkv_a_proj.quant_method, "quantization_config")
```
Comment (ping1jing2) [inline python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py:462]: from community code style: never use hasattr/getattr in sglang

File: python/sglang/srt/disaggregation/utils.py [hunk 2]
```diff
@@ -144,7 +144,7 @@ def __init__(
             )
             # Request validation: store bootstrap_room to detect metadata corruption
             self.bootstrap_room = torch.zeros(
-                (size, 8), dtype=torch.uint64, device=device
+                (size, 8), dtype=torch.int64, device=device
```
Comment (ping1jing2) [inline python/sglang/srt/disaggregation/utils.py]: did you check https://github.com/sgl-project/sglang/pull/19141 this PR? maybe we should give a solid reason for this change

**Review by RuixuanZhang06 — APPROVED** [review]

File: python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py (hunk 1 above)
Comment (MichelleWu351) [reply python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py:462]: The sglang repository has many lines written in this style.

**Review by iforgetmyname — APPROVED** [review]

File: python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py (hunk 1 above)
Comment (iforgetmyname) [reply python/sglang/srt/hardware_backend/npu/attention/mla_preprocess.py:462]: `quantization_config` is not a common member variable of `quant_method` so checking with `hasattr` is required.

The ultimate solution is to add this member variable into the base class but that will be   a huge work and shouldn't do such in this bugfix PR.

**Review by iforgetmyname — APPROVED** [review]

**Comment by iforgetmyname** [comment]
https://github.com/sgl-project/sglang/actions/runs/22480556278
https://github.com/sgl-project/sglang/actions/runs/22480556256

image

passed all gpu and npu ci tests
